src/metadata.js

```diff
@@ -7,7 +7,7 @@ import {
     pick
 } from 'lodash';
 import { ajax, get, post, del, parseJSON } from './xhr';
```

Comment:
Here I smell really bad code:
- `xhr.get()` already calls `r.json()` on its output (on the contrary from other methods POST/PUT/DEL)
- everywhere in this file `get()` is followed by `.then(r => (r.ok ? r.json() : r))`  (see [L152](https://github.com/gooddata/gooddata-js/pull/231/files#diff-7f3b58adf9e7b7e802f63cc9b3855506L152)), which means that if the resulting json had an "ok" field, it would call `r.json()` again and fail.
- also there could be problem with empty body, i am not sure how fetch handles `r.json()` on empty stream.

--> i suggest removing the `r.json()` in 'xhr'  (so it would be consistent with other methods)
--> the other option is to remove all the `.then(r => (r.ok ? r.json() : r))`  constructions

Reply:
Inconsistent but working, I would leave it for another task (so this PR could get merged)